src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -828,7 +838,7 @@ object GraphFrame extends Serializable with Logging {
         } else {
           val eRen = nestE(name)
           val dstV = nestV(dstName)
-          (Some(maybeJoin(prev, eRen)
+          (Some(maybeCrossJoin(prev, eRen)
```

Comment:
Good point.  I can't figure out why this used left_outer; it seems to have been that way from the beginning, so we probably never thought about it.

CC @ankurdave in case you're watching--do you recall? : )

Reply:
I don't remember either :)